test/TitleEscrow.js

```diff
@@ -1,129 +1,152 @@
-};
-const assertTransferOwnerApprovalLog = (log, sender, receiver) => {
-  expect(log.event).to.deep.equal("TransferOwnerApproval");
-  expect(log.args[1]).to.deep.equal(sender);
-  expect(log.args[2]).to.deep.equal(receiver);
-};
-const assertHolderChangedLog = (log, sender, receiver) => {
-  expect(log.event).to.deep.equal("HolderChanged");
-  expect(log.args[0]).to.deep.equal(sender);
-  expect(log.args[1]).to.deep.equal(receiver);
-};
-contract("TitleEscrow", accounts => {
-  const carrier1 = accounts[0];
-  const beneficiary1 = accounts[1];
-  const beneficiary2 = accounts[2];
-  const holder1 = accounts[3];
-  const holder2 = accounts[4];
+const { expect } = require("chai").use(require("chai-as-promised"));
+describe("TitleEscrow", async () => {
+  let accounts;
+  let TitleEscrowFactory;
+  let ERC721Factory;
+  let CalculateSelectorFactory;
+  let TitleEscrowCreatorFactory;
+
+  let carrier1, beneficiary1, beneficiary2, holder1, holder2;
+
+  before("", async () => {
+    accounts = await ethers.getSigners();
+    TitleEscrowFactory = await ethers.getContractFactory("TitleEscrow");
+    ERC721Factory = await ethers.getContractFactory("TradeTrustERC721");
+    CalculateSelectorFactory = await ethers.getContractFactory("CalculateSelector");
+    TitleEscrowCreatorFactory = await ethers.getContractFactory("TitleEscrowCreator");
+    [carrier1, beneficiary1, beneficiary2, holder1, holder2, ..._] = accounts;
+  });
+
+  const SAMPLE_TOKEN_ID = "0x624d0d7ae6f44d41d368d8280856dbaac6aa29fb3b35f45b80a7c1c90032eeb3";
+  const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+
+  const assertTransferLog = (logs, from, to) => {
+    expect(logs.event).to.deep.equal("Transfer");
+    expect(logs.args[0]).to.deep.equal(from);
+    expect(logs.args[1]).to.deep.equal(to);
+  };
+  const assertTitleReceivedLog = (log, tokenRegistry, sender) => {
+    expect(log.event).to.deep.equal("TitleReceived");
+    expect(log.args[0]).to.deep.equal(tokenRegistry);
+    expect(log.args[1]).to.deep.equal(sender);
+  };
+  const assertTitleCededLog = (log, tokenRegistry, receiver) => {
+    expect(log.event).to.deep.equal("TitleCeded");
+    expect(log.args[0]).to.deep.equal(tokenRegistry);
+    expect(log.args[1]).to.deep.equal(receiver);
+  };
+  const assertTransferOwnerApprovalLog = (log, sender, receiver) => {
+    expect(log.event).to.deep.equal("TransferOwnerApproval");
+    expect(log.args[1]).to.deep.equal(sender);
+    expect(log.args[2]).to.deep.equal(receiver);
+  };
+  const assertHolderChangedLog = (log, sender, receiver) => {
+    expect(log.event).to.deep.equal("HolderChanged");
+    expect(log.args[0]).to.deep.equal(sender);
+    expect(log.args[1]).to.deep.equal(receiver);
+  };
 
   let ERC721Address = "";
   let ERC721Instance;
 
-  beforeEach("", async () => {
-    ERC721Instance = await ERC721.new("foo", "bar");
+  beforeEach("doot", async () => {
+    ERC721Instance = await ERC721Factory.connect(carrier1).deploy("foo", "bar");
```

Comment:
i like how its using .connect() here with hardhat, easier to understand than adding the extra object at the back to indicate the account initiated the contract instance.